### ISSUE TITLE
LoadError: LoadError: UndefVarError: expand! not defined

### ISSUE THREAD

**Issue by tanders** [FluxML/Zygote.jl#206]
On a fresh installation of Julia 1.1.1 (2019-05-16), I ran into the following issue. Any help?

Edit: I also asked for help at https://discourse.julialang.org/t/zygote-load-error-on-recent-julia-version/24763

Thank you! 

Best,
Torsten

using Zygote
    > Failed to precompile Zygote

[ Info: Precompiling Zygote [e88e6eb3-aa80-5325-afca-941959d7151f]
ERROR: LoadError: LoadError: UndefVarError: expand! not defined
Stacktrace:
 [1] include at ./boot.jl:326 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1038
 [3] include at ./sysimg.jl:29 [inlined]
 [4] include(::String) at ~/.julia/packages/Zygote/8rrjD/src/Zygote.jl:1
 [5] top-level scope at none:0
 [6] include at ./boot.jl:326 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1038
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:328 [inlined]
 [11] eval(::Expr) at ./client.jl:404
 [12] top-level scope at ./none:3
in expression starting at ~.julia/packages/Zygote/8rrjD/src/compiler/reverse.jl:1
in expression starting at ~/.julia/packages/Zygote/8rrjD/src/Zygote.jl:23

**Comment by tpapp** [comment]
Can't replicate this.
```julia
julia> versioninfo()
Julia Version 1.1.1
Commit 55e36cc308 (2019-05-16 04:10 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-6560U CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_CMDSTAN_HOME = /home/tamas/src/cmdstan/

(v1) pkg> st Zygote
    Status `~/.julia/environments/v1/Project.toml`
  [f6369f11] ForwardDiff v0.10.3
  [7869d1d1] IRTools v0.2.1
  [1914dd2f] MacroTools v0.5.0
  [276daf66] SpecialFunctions v0.7.2
  [e88e6eb3] Zygote v0.3.1
  [37e2e46d] LinearAlgebra 
```

**Comment by tanders** [comment]
Thanks for your response. I tried this on OS X. I tried to provide the similar information as you did below:

```julia
julia> versioninfo()
Julia Version 1.1.1
Commit 55e36cc (2019-05-16 04:10 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin15.6.0)
  CPU: Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)

(trebuchet) pkg> st Zygote
    Status `~/Julia/LearningJulia/Flux/model-zoo/games/differentiable-programming/trebuchet/Project.toml`
  [e88e6eb3] Zygote v0.3.1 #master (https://github.com/FluxML/Zygote.jl.git)
[snip]

(trebuchet) pkg> st
    Status `~/Julia/LearningJulia/Flux/model-zoo/games/differentiable-programming/trebuchet/Project.toml`
  [fbb218c0] BSON v0.2.3
  [864edb3b] DataStructures v0.15.0
  [31c24e10] Distributions v0.20.0
  [587475ba] Flux v0.8.3
  [98b73d46] Trebuchet v0.1.0 #tolerance (https://github.com/MikeInnes/Trebuchet.jl)
  [0f1e0344] WebIO v0.8.1
  [e88e6eb3] Zygote v0.3.1 #master (https://github.com/FluxML/Zygote.jl.git)
  [de0858da] Printf 
  [9a3f8284] Random 
  [10745b16] Statistics 
```

**Comment by tpapp** [comment]
It looks like you are using `master` (for which I can replicate the error), not the released version 0.3.1.

If you cannot use the released version, then
```julia
pkg> add IRTools#master
```
seems to fix the issue. 

I was wondering why this was not caught by CI, I think because of 05a69e2. @Roger-luo, was using `IRTools#master` for CI intentional?

**Comment by tanders** [comment]
Thanks for your helpful feedback. I am just a newbie trying to get an example running...

What I did was in the trebuchet example run `update; precompile` as instructed by the Julia doc on loading projects (https://julialang.github.io/Pkg.jl/v1/environments/), which then seemingly tried to run the master version of Zygote. I checked the file `Manifest.toml` (see https://github.com/FluxML/model-zoo/blob/master/games/differentiable-programming/trebuchet/Manifest.toml), and it indeed specifies `repo-rev = "master"` for Zygote.

What I now did was changing locally the version of Zygote in `Manifest.toml` by commenting out the line `repo-rev = "master"`, and that compiles without any errors. I will see how the example works with that...

Thanks a lot for your help!
